components/ai_chat/core/browser/ai_chat_service.cc

```diff
@@ -0,0 +1,373 @@
+  // if we don't call SetAssociatedContentDelegate, the conversation still
+  // has a default Tab's navigation for which is is associated. The Conversation
+  // won't use that Tab's Page for context.
+  content_conversations_.insert_or_assign(
+      associated_content_id, conversation->get_conversation_uuid());
+
+  return conversation;
+}
+
+void AIChatService::MarkAgreementAccepted() {
+  SetUserOptedIn(profile_prefs_, true);
+}
+
+void AIChatService::GetActionMenuList(GetActionMenuListCallback callback) {
+  std::move(callback).Run(ai_chat::GetActionMenuList());
+}
+
+void AIChatService::GetPremiumStatus(GetPremiumStatusCallback callback) {
+  credential_manager_->GetPremiumStatus(
+      base::BindOnce(&AIChatService::OnPremiumStatusReceived,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void AIChatService::GetCanShowPremiumPrompt(
+    GetCanShowPremiumPromptCallback callback) {
+  bool has_user_dismissed_prompt =
+      profile_prefs_->GetBoolean(prefs::kUserDismissedPremiumPrompt);
+
+  if (has_user_dismissed_prompt) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  base::Time last_accepted_disclaimer =
+      profile_prefs_->GetTime(prefs::kLastAcceptedDisclaimer);
+
+  // Can't show if we haven't accepted disclaimer yet
+  if (last_accepted_disclaimer.is_null()) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  base::Time time_1_day_ago = base::Time::Now() - base::Days(1);
+  bool is_more_than_24h_since_last_seen =
+      last_accepted_disclaimer < time_1_day_ago;
+
+  if (is_more_than_24h_since_last_seen) {
+    std::move(callback).Run(true);
+    return;
+  }
+
+  std::move(callback).Run(false);
+}
+
+void AIChatService::DismissPremiumPrompt() {
+  profile_prefs_->SetBoolean(prefs::kUserDismissedPremiumPrompt, true);
+}
+
+void AIChatService::OnPremiumStatusReceived(GetPremiumStatusCallback callback,
+                                            mojom::PremiumStatus status,
+                                            mojom::PremiumInfoPtr info) {
+#if BUILDFLAG(IS_ANDROID)
+  // There is no UI for android to "refresh" with an iAP - we are likely still
+  // authenticating after first iAP, so we should show as active.
+  if (status == mojom::PremiumStatus::ActiveDisconnected &&
+      profile_prefs_->GetBoolean(prefs::kBraveChatSubscriptionActiveAndroid)) {
+    status = mojom::PremiumStatus::Active;
+  }
+#endif
+
+  last_premium_status_ = status;
+  if (ai_chat::HasUserOptedIn(profile_prefs_) && ai_chat_metrics_ != nullptr) {
+    ai_chat_metrics_->OnPremiumStatusUpdated(false, status, std::move(info));
+  }
+  model_service_->OnPremiumStatus(status);
+  std::move(callback).Run(status, std::move(info));
+}
+
+void AIChatService::MaybeEraseConversation(
+    ConversationHandler* conversation_handler) {
+  if (!conversation_handler->IsAnyClientConnected() &&
+      (!features::IsAIChatHistoryEnabled() ||
+       !conversation_handler->HasAnyHistory())) {
```

Comment:
Maybe want to include the case where all entries in history was staged entries?

Reply:
done c4bf8886b04